File: toolkit/cli/src/entity/mapper/api-information.mapper.ts
/*
 * Copyright (c) 2025 Timon Borter <timon.borter@gmx.ch>
 * Licensed under the Polyform Small Business License 1.0.0
 * See LICENSE file for full details.
 */

import type { CalculateQualityGateRequestIncludeApisInner } from '../../clients/quality-gate-api';
import type { ApiInformation } from '../../config/sanitized-options';

export const toDtos = (apiInformation: ApiInformation[]): CalculateQualityGateRequestIncludeApisInner[] => {
  return apiInformation.map(api => ({
    serviceName: api.serviceName,
    apiName: api.apiName,
    apiVersion: api.apiVersion,
  }));
};

File: microservices/api-sync-job/src/main/java/io/github/bbortt/snow/white/microservices/api/sync/job/domain/model/ApiInformationMapper.java
/*
 * Copyright (c) 2025 Timon Borter <timon.borter@gmx.ch>
 * Licensed under the Polyform Small Business License 1.0.0
 * See LICENSE file for full details.
 */

package io.github.bbortt.snow.white.microservices.api.sync.job.domain.model;

import static org.mapstruct.MappingConstants.ComponentModel.SPRING;

import io.github.bbortt.snow.white.microservices.api.sync.job.api.client.apiindexapi.dto.GetAllApis200ResponseInner;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = SPRING)
public interface ApiInformationMapper {
  @Mapping(target = "apiName", source = "name")
  @Mapping(target = "apiVersion", source = "version")
  GetAllApis200ResponseInner toDto(ApiInformation apiInformation);
}
